lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -245,43 +240,78 @@ export class UserAgentApplication {
       // TODO: Should we throw noCallback error here?
     }
 
+    // if extraScopesToConsent is passed, append them to the login request
+    let scopes: Array<string>;
+
+    if (request) {
+      if (request.extraScopesToConsent) {
+        scopes = [...request.scopes, ...request.extraScopesToConsent];
+      }
+      else {
+        scopes = request.scopes;
+      }
+    }
+
     // TODO: Replace with new validation pattern - This will come with Error
     if (scopes) {
       const isValidScope = this.validateInputScope(scopes);
       if (isValidScope && !Utils.isEmpty(isValidScope)) {
           if (this.tokenReceivedCallback) {
-              this.tokenReceivedCallback(ErrorDescription.inputScopesError, null, ErrorCodes.inputScopesError, Constants.idToken, this.getUserState(this.cacheStorage.getItem(Constants.stateLogin, this.inCookie)));
+            this.tokenReceivedCallback(ErrorDescription.inputScopesError, null, ErrorCodes.inputScopesError, Constants.idToken, this.getUserState(this.cacheStorage.getItem(Constants.stateLogin, this.inCookie)));
           return;
         }
       }
       scopes = this.filterScopes(scopes);
     }
 
-    // extract ADAL id_token if exists
-    let idTokenObject = this.extractADALIdToken();
+    let user = this.getUser();
 
-    // silent login if ADAL id_token is retrieved successfully - SSO
-    if (idTokenObject && !scopes) {
-      this.logger.info("ADAL's idToken exists. Extracting login information from ADAL's idToken ");
-      extraQueryParameters = Utils.constructUnifiedCacheExtraQueryParameter(idTokenObject, extraQueryParameters);
-      this.silentLogin = true;
-      this.acquireTokenSilent([this.clientId], this.authority, this.getUser(), extraQueryParameters)
-      .then((idToken) => {
-        this.silentLogin = false;
-        this.logger.info("Unified cache call is successful");
-        // TODO: Change callback to return AuthRespkjoonse
-        if (this.tokenReceivedCallback) {
-          this.tokenReceivedCallback.call(this, null, idToken, null, Constants.idToken, this.getUserState(this.silentAuthenticationState));
-        }
-      }, (error) => {
-        this.silentLogin = false;
-        this.logger.error("Error occurred during unified cache ATS");
-        this.loginRedirectHelper(scopes, extraQueryParameters);
-      });
+    let extraQueryParameters: KeyValue = null;
```

Comment:
you may want to rename extraQueryParameters to avoid confusion with request.extraQueryParameters

Reply:
Done. This is not longer required.